mbed_devices/_internal/resolve_target.py

```diff
@@ -17,60 +16,59 @@ class NoTargetForCandidate(Exception):
 
 
 def resolve_target(candidate: CandidateDevice) -> MbedTarget:
-    """Resolves target for given CandidateDevice if possible.
-
-    Currently, the mechanism for resolving targets relies on existence of HTM files in devices mass storage.
-    The specification of those HTM files is that they redirect to devices product page on os.mbed.com.
-    Information about Mbed Enabled requirements: https://www.mbed.com/en/about-mbed/mbed-enabled/requirements/
-    """
-    target_resolver = _build_target_resolver(candidate)
-
+    """Resolves target for given CandidateDevice if possible."""
     try:
-        return target_resolver()
+        return _resolve_target(candidate)
     except UnknownTarget:
         raise NoTargetForCandidate
 
 
-def _build_target_resolver(candidate: CandidateDevice) -> Callable:
-    all_files_contents = _get_all_htm_files_contents(candidate.mount_points)
-    try:
-        product_code = _extract_product_code(all_files_contents)
-        return functools.partial(get_target_by_product_code, product_code)
-    except ProductCodeNotFound:
-        pass
+def _resolve_target(candidate: CandidateDevice) -> MbedTarget:
+    """Resolves target for a given CandidateDevice.
 
-    try:
-        online_id = _extract_online_id(all_files_contents)
-        return functools.partial(get_target_by_online_id, slug=online_id.device_slug, target_type=online_id.device_type)
-    except OnlineIdNotFound:
-        pass
+    This function interrogates CandidateDevice, attempting to establish the best method to resolve an MbedTarget,
+    the rules are as follows:
 
-    # Product code might be the first 4 characters of the serial number
-    return functools.partial(get_target_by_product_code, candidate.serial_number[:4])
+    1. Use product code retrieved from one of HTM files in the mass storage if available.
+    2. Use online id retrieved from one of the HTM files in the mass storage if available.
+    3. Fallback to product code retrieved from serial number.
 
+    The specification of HTM files is that they redirect to devices product page on os.mbed.com.
+    Information about Mbed Enabled requirements: https://www.mbed.com/en/about-mbed/mbed-enabled/requirements/
+    """
+    all_files_contents = _get_all_htm_files_contents(candidate.mount_points)
+
+    # 1/
```

Comment:
Seems like overkill to have these comments here. I can read the code, which is fairly clear in this case.

Reply:
https://github.com/ARMmbed/mbed-devices/pull/51/commits/f115ce593c8a51029d9b957fe9a32970df3b371e